pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1724,7 +1724,7 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                                         });
                                                 persistentTopic.stopReplProducers()
                                                         .whenCompleteAsync((v, exception) -> {
-                                                            topics.remove(topic, topicFuture);
+                                                            persistentTopic.close();
```

Comment:
Aha, I wanted to fix it there before. 

If you chose the topic closely. You don't need the following codes. Because topic#close already does it.

```
 persistentTopic.getTransactionBuffer()
                                                        .closeAsync()
                                                        .exceptionally(t -> {
                                                            log.error("[{}] Close transactionBuffer failed", topic, t);
                                                            return null;
                                                         });
                                                 persistentTopic.stopReplProducers()
```

Reply:
Plus, please give good log information if the closing operation has errors. :)

Reply:
@mattisonchao 

> Aha, I wanted to fix it there before.

Yes, it is #21445 

> If you chose the topic close method. You don't need the following codes. Because topic#close already does it.
> Plus, please give good log information if the closing operation has errors. :)

Sure, fixed.